test/Ecosystem.jl

```diff
@@ -0,0 +1,139 @@
+using Test
+using ADRIA
+using ADRIA.Distributions
+
+@testset "Truncated normal mean" begin
+
+    means::Vector{Float64} = Vector{Float64}([0.0, 0.5, 1.5, 2.5, 4.0, 5.0])
+    stdev::Vector{Float64} = Vector{Float64}([0.1, 0.25, 0.5, 1.0, 2.0, 5.0, 10.0])
+    lower_bounds::Vector{Float64} = Vector{Float64}([0.0, 0.5, 1.5, 3.0, 7.0, 10.0])
+    bound_widths::Vector{Float64} = Vector{Float64}([0.25, 0.5, 1.0, 2.0, 5.0, 10.0])
+
+    n_checks = length(means) * length(stdev) * length(bound_widths)
+    mean_diffs::Vector{Float64} = Vector{Float64}(undef, n_checks)
+
+    # truncated normal mean should agree with normal mean for symmetrical bounds about mean
+    ind::Integer = 1;
+    for mu in means
+        for std in stdev
+            for width in bound_widths
+                calculated = ADRIA.truncated_normal_mean(
+                    mu, std, mu - width, mu + width
+                )
+                mean_diffs[ind] = abs(mu - calculated)
+                ind += 1
+            end
+        end
+    end
```

Comment:
Nesting like this is usually a code smell. Use iterators instead.

I think you want something like this:

```julia
for (mu, std, width) in Iterators.product(means, stdev, bound_widths)
    @info (mu, std, width)
end
```

FYI, `std` is a problematic choice of variable name as it's also a function name (`std()` from Statistics.jl).

I've used `stdev` instead for clarity. Suggest changing the current `stdev` to `stdevs` (note the `s` to pluralize).

Aside from that, I'm not sure what `ind` is for?